webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/ProjectsLegalInfoRestService.scala

```diff
@@ -0,0 +1,153 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.admin.api.service
+
+import cats.syntax.traverse.*
+import org.eclipse.rdf4j.sparqlbuilder.rdf.Rdf
+import zio.IO
+import zio.Task
+import zio.ZIO
+import zio.ZLayer
+import zio.json.JsonCodec
+
+import dsp.errors.ForbiddenException
+import dsp.errors.InconsistentRepositoryDataException
+import dsp.errors.NotFoundException
+import org.knora.webapi.slice.admin.api.CopyrightHolderAddRequest
+import org.knora.webapi.slice.admin.api.CopyrightHolderReplaceRequest
+import org.knora.webapi.slice.admin.api.LicenseDto
+import org.knora.webapi.slice.admin.api.model.FilterAndOrder
+import org.knora.webapi.slice.admin.api.model.PageAndSize
+import org.knora.webapi.slice.admin.api.model.PagedResponse
+import org.knora.webapi.slice.admin.domain.model.Authorship
+import org.knora.webapi.slice.admin.domain.model.CopyrightHolder
+import org.knora.webapi.slice.admin.domain.model.KnoraProject
+import org.knora.webapi.slice.admin.domain.model.KnoraProject.Shortcode
+import org.knora.webapi.slice.admin.domain.model.User
+import org.knora.webapi.slice.admin.domain.service.KnoraProjectService
+import org.knora.webapi.slice.admin.domain.service.LicenseService
+import org.knora.webapi.slice.admin.domain.service.ProjectService
+import org.knora.webapi.slice.common.api.AuthorizationRestService
+import org.knora.webapi.slice.common.repo.rdf.Vocabulary
+import org.knora.webapi.store.triplestore.api.TriplestoreService
+import org.knora.webapi.store.triplestore.api.TriplestoreService.Queries.Select
+
+final case class ProjectsLegalInfoRestService(
+  private val licenses: LicenseService,
+  private val projects: KnoraProjectService,
+  private val auth: AuthorizationRestService,
+  private val triplestore: TriplestoreService,
+) {
+
+  def findAuthorships(user: User)(
+    shortcode: Shortcode,
+    pageAndSize: PageAndSize,
+    filterAndOrder: FilterAndOrder,
+  ): Task[PagedResponse[Authorship]] =
+    for {
+      project         <- auth.ensureProjectMember(user, shortcode)
+      totalAndAuthors <- queryAuthorships(project, pageAndSize, filterAndOrder)
+      (total, authors) = totalAndAuthors
+    } yield PagedResponse.from(authors, total, pageAndSize)
+
+  private def queryAuthorships(
+    project: KnoraProject,
+    paging: PageAndSize,
+    filterAndOrder: FilterAndOrder,
+  ): Task[(Int, Seq[Authorship])] = {
+    val graph                 = Rdf.iri(ProjectService.projectDataNamedGraphV2(project).value)
+    val searchTermQueryString = filterAndOrder.filter.map(Rdf.literalOf).map(_.getQueryString)
+    val authorVar             = "author"
+    val graphPattern =
+      s"""GRAPH ${graph.getQueryString} {
+         |  ?fileValue ${Vocabulary.KnoraBase.hasAuthorship.getQueryString} ?$authorVar .
+         |  ${searchTermQueryString.fold("")(term => s"FILTER(CONTAINS(LCASE(STR(?$authorVar)), ${term.toLowerCase}))")}
+         |}""".stripMargin
```

Comment:
Question: how about using the query builder? Would that over-complicate things, or make it more readable?

Reply:
I started with using the rdf4j sparq builder and even use parts of it for the escaping of the `searchtTermQueryString`. Unfortunately it lacks support for the whole `FILTER(CONTAINS(LCASE(STR(...))))` expression.

Reply:
fair enough, too bad that it doesn't support the filtering